project/src/main/java/com/google/sps/data/VolunteeringOpportunity.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps.data;
+
+import java.util.Set;
+
+public final class VolunteeringOpportunity {
+  private String name;
+  private int numSpotsLeft;
+  private Set<String> requiredSkills;
+  private Set<String> volunteers;
+
+  public static class Builder {
+    // Required parameters
+    private String name;
+    private int numSpotsLeft;
+
+    // Optional parameters
+    private Set<String> requiredSkills;
+    private Set<String> volunteers;
+
+    public Builder(String name, int numSpotsLeft) {
+      this.name = name;
+      this.numSpotsLeft = numSpotsLeft;
+    }
+
+    public Builder requiredSkills(Set<String> requiredSkills) {
+      this.requiredSkills = requiredSkills;
```

Comment:
I know the Builder allows fields to be null until set but do you somehow need to initialize requireSkills = new HashSet<>(); first? Or just let it be overwritten by Set<String> requiredSkills each time?

Reply:
Done.